### ISSUE TITLE
Trying to emulate IP cameras

### ISSUE THREAD

**Issue by josephKhoury95** [attify/firmware-analysis-toolkit#59]
Hello,

I am interested in using firmware-analysis-toolkit to emulate IP cameras, is it possible using this tool?
If yes can you please suggest some firmware to test, because I am still not able to emulate IP cameras...

Thank you

**Comment by extremecoders-re** [comment]
It should be possible to firmware emulate of IP camera. However it's not always a straightforward process.

Did you try to extract the firmware using binwalk?

**Comment by josephKhoury95** [comment]
Yes, I know it is not a straight forward process.

Yes, I did use Binwalk to extract the firmware binary file. But also Attify has Binwalk in its steps, right?

@extremecoders-re, since you are a contributor to this project. I also have a question for you. Is it possible to emulate multiple firmware at the same time using Attify? I know I need to edit a lot of files to make it work, but is it possible from your point of view?

Thank you

**Comment by extremecoders-re** [comment]
Yes, firmadyne and fat does use binwalk. My point was to try out binwalk separately and check the files it extracts. If binwalk doesn't work proper in the first place, firmadyne won't work too.

Emulating multiple firmware's is possible. Each firmware get's a different tap interface. As long as this condition is satisfied it should work.

To test, you can also try emulating the same firmware twice at the same time.  FAT doesn't use the SQL database, so they will be assigned different ID's even if they are the same files.

**Comment by extremecoders-re** [comment]
Another possible issue that just came to my mind is conflict of IP address. By default most of the firmware use the `192.168.0.0/24` address. Firmadyne set's up a route so that the traffic gets routed properly to the tap interface. The same address cannot route to multiple interfaces. This issue has to be sorted out.
